### ISSUE TITLE
BulkRead with ReplaceReadEntities=True Not Working as Expected When No Matches Found

### ISSUE THREAD

**Issue by champin4** [borisdj/EFCore.BulkExtensions#1637]
Long story short is I would expect if ReplaceReadEntities = true for the BulkConfig options for a .BulkRead(Async) that I would end up with an empty list if there were no matches on the database.

Right now we are having to rely on a specific field we are 99% sure should never be a default value to validate if the entities were actually replaced, however this is not optimal and would not work for all use cases.

This is with a SQL Server database if that matters.

**Comment by borisdj** [comment]
So you would like for list to be cleared if none are matched?

**Comment by champin4** [comment]
Yes that seems to be in line with what I see happening when there are matches and ReplaceReadEntities = true. An empty list would signify there were no matches just like a N sized list indicates there were N matches

**Comment by alexander-shustov** [comment]
It looks like the reason is that a new list `entitiesObjects` is being created in this row and it is also being cleared. The original list `entities` remains unchanged.
https://github.com/borisdj/EFCore.BulkExtensions/blob/2888e5284a2967b89de2c4ceaa0c06a716e7da74/EFCore.BulkExtensions.Core/TableInfo.cs#L982
